teacher.rb

```diff
@@ -0,0 +1,14 @@
+require './person'
+# Teacher's information/actions in the library.
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(age, name, specialization)
+    super
+    @specialization = specialization
+  end
```

Comment:
- [ ] I would recommend you refactor your method to this to allow creation of a teacher instance:

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

Reply:
Hello @TedLivist, can you explain what is the _super_ functionality in this case? I thought that by putting _ super_ I would re-use the parameters from the parent class.

Reply:
Hi @Gopxfs you're right in some way. That works IF the number of arguments in the child class is the same as the number of arguments in the parent class for the same method. This code below would work for example:

```
def initialize(age, name = 'Unknow', parent_permission: true)
  super
  @specialization = specialization
end
```
But that would mean we have to ignore the `specialization` attribute, which should not be the case. And since we cannot ignore it, we have to call `super()` with the selected arguments we wish to pass up to the parent class hence the suggested code above.

You can refer to [this article for more details](https://www.rubyguides.com/2018/09/ruby-super-keyword/).

Reply:
Ok this was very clear, thanks so much!